### PR TITLE
Temp solution done.

### PR DESCRIPTION
**Pull request by aliaksandr-galgouski**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -5,11 +5,13 @@
     "es6": false
   },
   "extends": ["eslint:recommended", "plugin:prettier/recommended"],
+  "parser": "babel-eslint",
   "parserOptions": {
     "ecmaVersion": 8,
-    "sourceType": "module"    
+    "sourceType": "module"
```
Comment (AlexanderGolub) [inline .eslintrc]: pay attention to those unnecessary changes

File: src/app.js [hunk 2]
```diff
@@ -1,65 +1,24 @@
...
+
+import NewsApiService from "./services/news-api.service";
 
 const API_KEY = "cd77286a82d24223b1320716435aacb6";
 
 export default class App {
-  constructor() {
-    this.service = new NewsApiService(API_KEY);
-  }
+  constructor() {}
```
Comment (AlexanderGolub) [inline src/app.js]: you can omit empty constructor

File: src/services/http.factory.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const DEFAULT_HEADERS = {
+  "Accept": "application/json"
+};
+
+function doRequest(method, url, headers, body) {
+  let init = {};
+  init.url = url;
+  init.body = body;
+  init.method = method;
+  init.headers = Object.assign({}, DEFAULT_HEADERS, headers);
```
Comment (AlexanderGolub) [inline src/services/http.factory.js]: use ES6 features `{ ...DEFAULT_HEADERS, ...headers }`

File: src/interceptors/trace.interceptor.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const handler = {    
+  get(target, propKey, receiver) {
+    const origMethod = target[propKey];
+
+    return function(...args) {
+      console.debug("Invoked", propKey);
+      console.debug("With arguments:", args);
+
+      let result = origMethod.apply(this, args);
```
Comment (AlexanderGolub) [inline src/interceptors/trace.interceptor.js]: `result` is `const`

File: src/services/news-api.service.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  async getSources() {
+    const url = `${BASE_URL}/sources`;
+    const request = Http.get(url, this.headers);
+    const json = await handleErrors(request);
+
+    return (json && json.sources) || [];
+  }
+
+  async getNews(sourceId, sortBy) {
+    sortBy = sortBy || "latest";
```
Comment (AlexanderGolub) [inline src/services/news-api.service.js:49]: missing `const`

File: src/services/notification.service.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+        <strong>Error:</strong> ${message.toString()}
+        <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>`;
+
+    this.$body.append(html);
+  }
+}
+NotificationService.instance = new NotificationService();
```
Comment (AlexanderGolub) [inline src/services/notification.service.js]: you can just `export default new NotificationService`

File: src/views/base.view.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+export default class View extends EventEmiter {
+  constructor($parent) {
+    super();
+    this.$parent = $parent;
+  }
+
+  render(model) {}
+  clearParent() {
+    this.$parent.html(String.empty);
```
Comment (AlexanderGolub) [inline src/views/base.view.js]: there is no such thing as `String.empty`, use `''`

File: src/views/elements/base.list.view.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  render(model) {
+    this.clearParent();
+    const $container = this.createContainer(model);
+
+    $container.on(
+      "click",
+      "> *",
+      function(e) {
```
Comment (AlexanderGolub) [inline src/views/elements/base.list.view.js]: use ES6 arrow functions, no need to `bind`

File: src/views/elements/base.list.view.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        this.emit("itemClicked", e.currentTarget);
+      }.bind(this)
+    );
+
+    (model || []).forEach(itemModel => this.renderItem($container, itemModel));
```
Comment (AlexanderGolub) [inline src/views/elements/base.list.view.js]: use `render(model = [])` if you need default parameter

File: src/services/news-api.service.js (hunk 5 above)
Comment (aliaksandr-galgouski) [reply src/services/news-api.service.js:49]: sortBy is a method argument

File: src/services/http.factory.js (hunk 3 above)
Comment (aliaksandr-galgouski) [reply src/services/http.factory.js]: replaced

File: src/interceptors/trace.interceptor.js (hunk 4 above)
Comment (aliaksandr-galgouski) [reply src/interceptors/trace.interceptor.js]: done

File: src/app.js (hunk 2 above)
Comment (aliaksandr-galgouski) [reply src/app.js]: removed

File: src/services/notification.service.js (hunk 6 above)
Comment (aliaksandr-galgouski) [reply src/services/notification.service.js]: replaced

File: src/views/base.view.js (hunk 7 above)
Comment (aliaksandr-galgouski) [reply src/views/base.view.js]: wiped

File: src/views/elements/base.list.view.js (hunk 8 above)
Comment (aliaksandr-galgouski) [reply src/views/elements/base.list.view.js]: replaced

File: src/views/elements/base.list.view.js (hunk 9 above)
Comment (aliaksandr-galgouski) [reply src/views/elements/base.list.view.js]: done
